library-best-practices/JLBP-18.md

```diff
@@ -0,0 +1,49 @@
+[JLBP-18] Use shading sparingly and carefully
+---------------------------------------------
+
+The top answer to the question at
+https://softwareengineering.stackexchange.com/questions/297276/what-is-a-shaded-java-dependency
```

Comment:
I would prefer this to be more self-contsined.

Reply:
That sounds reasonable, I'll make it self-contained.

Reply:
OK, I'd still rewrite this though. "Sparingly" is a bit stilted and old-fashioned. "Carefully" is OK but when would we not recommend being careful. Can we be more specific here? Maybe even "Shade dependencies only as a last resort" or "Use shading only if multiple conflicting versions of a dependency are required"